src/HotChocolate/Spatial/src/Types.Spatial/GeoJSONPositionScalar.cs

```diff
@@ -0,0 +1,225 @@
+﻿using System;
+using System.Collections;
+using HotChocolate.Language;
+using HotChocolate.Types;
+using NetTopologySuite.Geometries;
+
+namespace Types.Spatial
+{
+    public class GeoJSONPositionScalar : ScalarType<Coordinate>
+    {
+        /// https://tools.ietf.org/html/rfc7946#section-3.1.1
+        public GeoJSONPositionScalar() : base("Position")
+        {
+            Description = "A position is an array of numbers. There MUST be two or more elements.";
+        }
+
+        public override bool IsInstanceOfType(IValueNode literal)
+        {
+            if (literal == null)
+            {
+                throw new ArgumentNullException(nameof(literal));
+            }
+
+            if (literal is NullValueNode)
+            {
+                return true;
+            }
+
+            if (literal is ListValueNode listValueNode)
+            {
+                for (var i = 0; i < listValueNode.Items.Count; i++)
+                {
+                    if (listValueNode.Items[i].Value is ListValueNode || listValueNode.Items[i].Value is IList)
+                    {
+                        return false;
+                    }
+                }
+
+                return true;
+            }
+
+            return false;
+}
+
+        public override object ParseLiteral(IValueNode literal)
+        {
+            // parse literal from input (ListValueNode) into Coordinate
+
+            if (literal == null)
+            {
+                throw new ArgumentNullException(nameof(literal));
+            }
+
+            if (literal is NullValueNode)
+            {
+                return null;
+            }
+
+            if (literal is ListValueNode listNode)
+            {
+                if (listNode.Items.Count != 2 && listNode.Items.Count != 3)
+                {
+                    throw new ArgumentException(
+                        "The Position type has to have two or three elements [x,y] or [x,y,z]",
+                        nameof(literal));
+                }
+
+                var xNode = listNode.Items[0];
+                var yNode = listNode.Items[1];
+                if (xNode == null)
+                {
+                    throw new ArgumentException(
+                        "The first element of the array can not be null",
+                        nameof(literal));
+                }
+
+                if (yNode == null)
+                {
+                    throw new ArgumentException(
+                        "The second element of the array can not be null",
+                        nameof(literal));
+                }
+
+                var xValue = xNode switch
+                {
+                    FloatValueNode node => node.ToDouble(),
+                    IntValueNode node => node.ToDouble(),
```

Comment:
we have an interface for that `IFloatValueLiteral`. Both `IntValueNode` and `FloatValueNode` implement `IFloatValueLiteral` which has `ToDouble()`. So we can easily remove that switch and do a simplified check.

Reply:
i believe you already did this.